interop_binaries/src/bin/janus_interop_collector.rs

```diff
@@ -0,0 +1,503 @@
+                &collect_job_state.agg_param,
+            )
+            .context("could not decode aggregation parameter")?;
+            let vdaf =
+                Prio3::new_aes128_count(2).context("failed to construct Prio3Aes128Count VDAF")?;
+            let aggregate_result = vdaf
+                .unshard(&(), [leader_aggregate_share, helper_aggregate_share])
+                .context("could not unshard aggregate result")?;
+            Ok(Some(AggregationResult::Number(aggregate_result)))
+        }
+        VdafObject::Prio3Aes128Sum { bits } => {
+            let leader_aggregate_share =
+                AggregateShare::<Field128>::try_from(leader_aggregate_share_bytes.as_ref())
+                    .context("could not decode leader's aggregate share")?;
+            let helper_aggregate_share =
+                AggregateShare::<Field128>::try_from(helper_aggregate_share_bytes.as_ref())
+                    .context("could not decode helper's aggregate share")?;
+            <<Prio3Aes128Sum as Vdaf>::AggregationParam>::get_decoded(&collect_job_state.agg_param)
+                .context("could not decode aggregation parameter")?;
+            let vdaf = Prio3::new_aes128_sum(2, bits)
+                .context("failed to construct Prio3Aes128Sum VDAF")?;
+            let aggregate_result = vdaf
+                .unshard(&(), [leader_aggregate_share, helper_aggregate_share])
+                .context("could not unshard aggregate result")?;
+            Ok(Some(AggregationResult::Number(
+                aggregate_result
+                    .try_into()
+                    .context("aggregate result was too large to represent natively in JSON")?,
+            )))
+        }
+        VdafObject::Prio3Aes128Histogram { ref buckets } => {
+            let leader_aggregate_share =
+                AggregateShare::<Field128>::try_from(leader_aggregate_share_bytes.as_ref())
+                    .context("could not decode leader's aggregate share")?;
+            let helper_aggregate_share =
+                AggregateShare::<Field128>::try_from(helper_aggregate_share_bytes.as_ref())
+                    .context("could not decode helper's aggregate share")?;
+            <<Prio3Aes128Histogram as Vdaf>::AggregationParam>::get_decoded(
+                &collect_job_state.agg_param,
+            )
+            .context("could not decode aggregation parameter")?;
+            let vdaf = Prio3::new_aes128_histogram(2, buckets)
+                .context("failed to construct Prio3Aes128Histogram VDAF")?;
+            let aggregate_result = vdaf
+                .unshard(&(), [leader_aggregate_share, helper_aggregate_share])
+                .context("could not unshard aggregate result")?;
+            let converted = aggregate_result
+                .into_iter()
+                .map(|counter| {
+                    u64::try_from(counter).context(
+                        "entry in aggregate result was too large to represent natively in JSON",
```

Comment:
JSON numbers are floating point, which should be big enough to represent `u128`. I'm not sure if `u128` is big enough that loss of precision can occur, though.

Reply:
JSON numbers can only "safely" represent integers up to about 53 bits[1]. If we care to fix this (maybe not for now, since this is test code & can simply arrange to not use such large numbers), the standard method I've seen is to represent the integer as a JSON string like `"12345"`.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER